src/jq/api/api_impl.clj

```diff
@@ -95,6 +85,20 @@
   ^String [^JsonNode data ^JsonQuery query ^Scope scope]
   (json-node->string (apply-json-query-on-json-node data query scope)))
 
+(defn new-scope
+  (^Scope [] (Scope/newChildScope root-scope))
+  (^Scope [opts]
+   (let [^Scope scope (new-scope)
+         module-paths (get opts :modules)
+         module-paths (if (string? module-paths) [module-paths] module-paths)
+         variables (get opts :vars)]
+     (when (seq module-paths)
+       (setup-modules! scope module-paths))
+     (when variables
+       (doseq [[key value] variables]
+        (.setValue scope (name key) (if (string? value) (string->json-node value) value))))
```

Comment:
The value conversion is very naive, only string are supported.

Something like
```
(defn ->JsonNode ^JsonNode [data]
  (.valueToTree mapper data))
```
covers more cases. A notable case that is not handled is keywords. 

If a mapper would be parameterized, then a special one with support for keywords could be made.

Reply:
_nod_; this seems like an excellent use for Clojure's protocols, so `extend` could be used to add support for arbitrary types.

Reply:
...that said, if we were going to have a protocol for "anything that can be converted to a JsonNode", it would make sense to use that consistently for _all_ inputs, not just for `:vars`, so I'm not sure that it needs to be fit inside this particular PR.

Reply:
I agree, let's leave out the protocol part.